flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetSort.scala

```diff
@@ -71,6 +72,21 @@ class DataSetSort(
     )
   }
 
+  override def estimateRowCount(metadata: RelMetadataQuery): Double = {
+    val inputRowCnt = metadata.getRowCount(this.getInput)
+    if (inputRowCnt == null) {
+      inputRowCnt
+    } else {
+      val rowCount = Math.max(inputRowCnt - limitStart, 0D)
```

Comment:
inputRowCount might also just be an estimate and not guaranteed to be precise. Returning 1 is more robust, because it does not result in no-cost operators downstream.